examples/rectangle_area.py

```diff
@@ -0,0 +1,179 @@
+    def __repr__(self):
+        return "Rectangle: {height} X {width}".format(height=self.height, width=self.width)
+
+
+"""
+Next, we create a function to converts something into a Rectangle if possible.
+This function will be the coercion function for our custom PushType.
+"""
+
+
+def to_rect(thing):
+    return Rectangle(float(thing[0]), float(thing[1]))
+
+
+"""
+Next, we create the target function we would like to synthesize.
+Our target function returns the difference in area between two rectangles.
+This is also the function used to define of one of our custom instructions.
+"""
+
+
+def area_difference(r1, r2):
+    area_1 = r1.height * r1.width
+    area_2 = r2.height * r2.width
+    return (area_1 - area_2),  # comma at the end packages return value into a tuple
+
+
+"""
+Now we create another function used to define of one of our custom instructions.
+This function returns a tuple containing a rectangle made from two floats.
+"""
+
+
+def rectangle_from_floats(i1, i2):
+    return Rectangle(i1, i2),  # comma at the end packages return value into a tuple
+
+
+"""
+Next, we define our custom instructions that manipulate rectangles by defining parameters in SimpleInstruction.
+We want rectangle_areas_instruction to find difference in area between two rectangles, and push it to the float stack.
+We want rectangle_from_floats_instruction to take two floats and push a rectangle.
+"""
+
+rectangle_areas_instruction = SimpleInstruction(
+    name="rectangle_area_diff",  # unique name for the instruction
+    f=area_difference,  # function whose signature matches input_stacks and output_stacks
+    input_stacks=["rectangle", "rectangle"],  # list of PushType names to use when popping arguments from PushState
+    output_stacks=["float"],  # list of PushType names to use when pushing function results to the PushState
+    code_blocks=0  # number of CodeBlocks to open following the instruction in a Genome
+)
+
+rectangle_from_floats_instruction = SimpleInstruction(
+    name="rectangle_from_floats", f=rectangle_from_floats,
+    input_stacks=["float", "float"], output_stacks=["rectangle"], code_blocks=0
+)
+
+"""
+Now we must define our problem by creating two data sets of example input-output pairs.
+One will be used for training, and the other will be used to test our program on unseen data.
+To get the true output for the data sets, we will use the "target_function" function.
+"""
+
+X = [[Rectangle(row[0], row[1]), Rectangle(row[2], row[3])] for row in np.random.rand(20, 4)]
+y = [[area_difference(r[0], r[1])] for r in X]
+
+"""
+Next, we create a type library that specifies we will be synthesizing programs that manipulate 
+    "floats" (built-in to pyshgp) and "rectangles" (custom for this problem).
+"""
+
+type_library = (
+    PushTypeLibrary(register_core=False)
+        .register(PushFloat)
+        .create_and_register(name="rectangle", underlying_types=(Rectangle,), coercion_func=to_rect)
+)
```

Comment:
Very minor and nit-picky suggestion: Although many text editors get this wrong (including the one I use), the python style guides suggest vertical alignment when using`()` for a multi-line statement. For example:

```py
type_library = (
    PushTypeLibrary(register_core=False)
    .register(PushFloat)
    .create_and_register(name="rectangle", underlying_types=(Rectangle,), coercion_func=to_rect)
)
```